src/miner.cpp

```diff
@@ -74,321 +74,406 @@ int64_t UpdateTime(CBlockHeader* pblock, const Consensus::Params& consensusParam
     return nNewTime - nOldTime;
 }
 
-CBlockTemplate* CreateNewBlock(const CChainParams& chainparams, const CScript& scriptPubKeyIn,bool blockstreamCoreCompatible);
+BlockAssembler::BlockAssembler(const CChainParams &_chainparams) : chainparams(_chainparams)
```

Comment:
Please initialize all your members in the constructor.

pblock, nBlockSize, nBlockTx, nBlockSigOps, nFees, nHeight, nLockTimeCutoff, lasFewTxs, blockFinished were all forgotten here.

I didn't check very carefully, but are you sure all of those are actually useful at class scope? I have the strong impression half of them can be limited to only exist in the method they were used.

Please also remove the CChainParams instance, just use the global static method Params() when you need it.